src/ssl.c

```diff
@@ -21582,9 +22675,18 @@ int wolfSSL_RSA_LoadDer(WOLFSSL_RSA* rsa, const unsigned char* derBuf, int derSz
         return SSL_FATAL_ERROR;
     }
 
-    ret = wc_RsaPrivateKeyDecode(derBuf, &idx, (RsaKey*)rsa->internal, derSz);
+    if(opt == WOLFSSL_RSA_LOAD_PRIVATE)
+        ret = wc_RsaPrivateKeyDecode(derBuf, &idx, (RsaKey*)rsa->internal, derSz);
+    else
+        ret = wc_RsaPublicKeyDecode(derBuf, &idx, (RsaKey*)rsa->internal, derSz);
```

Comment:
With public key usage in wolfSSL_RSA_LoadDer_ex check on SetRsaExternal with the case of having only n and e.